eras/babbage/impl/src/Cardano/Ledger/Babbage/Translation.hs

```diff
@@ -0,0 +1,193 @@
+-- 'TranslationError', i.e., 'Void', it means the consensus layer relies on it
+-- being total. Do not change it!
+--------------------------------------------------------------------------------
+
+type instance PreviousEra (BabbageEra c) = AlonzoEra c
+
+type instance TranslationContext (BabbageEra c) = AlonzoGenesis
+
+instance
+  (Crypto c) =>
+  TranslateEra (BabbageEra c) NewEpochState
+  where
+  translateEra ctxt nes =
+    pure $
+      NewEpochState
+        { nesEL = nesEL nes,
+          nesBprev = nesBprev nes,
+          nesBcur = nesBcur nes,
+          nesEs = translateEra' ctxt $ nesEs nes,
+          nesRu = nesRu nes,
+          nesPd = nesPd nes
+        }
+
+instance Crypto c => TranslateEra (BabbageEra c) ShelleyGenesis where
+  translateEra ctxt genesis =
+    pure
+      API.ShelleyGenesis
+        { API.sgSystemStart = API.sgSystemStart genesis,
+          API.sgNetworkMagic = API.sgNetworkMagic genesis,
+          API.sgNetworkId = API.sgNetworkId genesis,
+          API.sgActiveSlotsCoeff = API.sgActiveSlotsCoeff genesis,
+          API.sgSecurityParam = API.sgSecurityParam genesis,
+          API.sgEpochLength = API.sgEpochLength genesis,
+          API.sgSlotsPerKESPeriod = API.sgSlotsPerKESPeriod genesis,
+          API.sgMaxKESEvolutions = API.sgMaxKESEvolutions genesis,
+          API.sgSlotLength = API.sgSlotLength genesis,
+          API.sgUpdateQuorum = API.sgUpdateQuorum genesis,
+          API.sgMaxLovelaceSupply = API.sgMaxLovelaceSupply genesis,
+          API.sgProtocolParams = translateEra' ctxt (API.sgProtocolParams genesis),
+          API.sgGenDelegs = API.sgGenDelegs genesis,
+          API.sgInitialFunds = API.sgInitialFunds genesis,
+          API.sgStaking = API.sgStaking genesis
+        }
+
+newtype Tx era = Tx {unTx :: Core.Tx era}
+
+instance
+  ( Crypto c,
+    Core.Tx (BabbageEra c) ~ ValidatedTx (BabbageEra c)
+  ) =>
+  TranslateEra (BabbageEra c) Tx
+  where
+  type TranslationError (BabbageEra c) Tx = DecoderError
+  translateEra _ctxt (Tx tx) = do
+    -- Note that this does not preserve the hidden bytes field of the transaction.
+    -- This is under the premise that this is irrelevant for TxInBlocks, which are
+    -- not transmitted as contiguous chunks.
+    bdy <- translateViaCBORAnn "txbody" $ Alonzo.body tx
+    txwits <- translateViaCBORAnn "txwitness" $ Alonzo.wits tx
+    -- transactions from Mary era always pass script ("phase 2") validation
+    aux <- case Alonzo.auxiliaryData tx of
+      SNothing -> pure SNothing
+      SJust axd -> SJust <$> translateViaCBORAnn "auxiliarydata" axd
+    let validating = IsValid True
+    pure $ Tx $ ValidatedTx bdy txwits validating aux
+
+--------------------------------------------------------------------------------
+-- Auxiliary instances and functions
+--------------------------------------------------------------------------------
+
+translateViaCBORAnn :: (ToCBOR a, FromCBOR (Annotator b)) => Text -> a -> Except DecoderError b
```

Comment:
I am trying to think of a good place for this function to live so that we don't have to copy it every era. Maybe in `Cardano.Ledger.Serialization` from the `cardano-ledger-core` package?